src/components/PrimaryNav.js

```diff
@@ -28,12 +29,31 @@ const PrimaryNav = forwardRef(
     const theme = useTheme();
     const toUpperCase = text => text.toUpperCase();
     const { isOpen, onOpen, onClose } = useDisclosure();
+    const location = useLocation();
     const handleToggle = () => {
-      // TODO: Prevent scrolling while the slideout is open. The below logic works, but if a user clicks a link they'll keep the fixed position
-      // document.body.style.position = isVisible ? 'unset' : 'fixed';
+      // If the sidenav is open then on its toggle we set the body position: fixed
+      document.body.style.position = 'fixed';
       setIsVisible(!isVisible);
     };
 
+    useEffect(() => {
+      // Handle the first render when the location state will be null.
+      if (!isVisible && location.state === null) {
+        document.body.style.position = 'unset';
+      }
+    }, []);
+
+    useEffect(() => {
+      // if the sidenav is open and the pages path does not match the previous path then we unset the position: fixed on the body
+      if (
+        !isVisible &&
+        location.state !== null &&
+        location.pathname !== location.state.referrer
+      ) {
+        document.body.style.position = 'unset';
+      }
+    }, [isVisible]);
```

Comment:
These may not change, but should be included in the dependency array here, I believe

```suggestion
    }, [isVisible, location.pathname, location.state]);
```